src/SAWScript/Crucible/LLVM/Builtins.hs

```diff
@@ -2227,6 +2236,13 @@ constructExpandedSetupValue cc sc loc t =
     Crucible.MetadataType   -> failUnsupportedType "Metadata"
     Crucible.VecType{}      -> failUnsupportedType "Vec"
     Crucible.X86_FP80Type{} -> failUnsupportedType "X86_FP80"
+    -- This case is a bit suspect, but it _should_ be unreachable under the
+    -- assumption that a SAW LLVMType value cannot represent an opaque pointer.
+    -- I (RGS) believe this to be the case, as the only way to construct a
+    -- pointer LLVMType at the moment is to use `llvm_pointer`, which constructs
+    -- a non-opaque pointer type. If we ever added something like, say,
+    -- `llvm_pointer_opaque`, we would need to reconsider this case.
+    Crucible.PtrOpaqueType  -> failUnsupportedType "PtrOpaque"
```

Comment:
It turns out my comment here is wrong, as you _can_ construct an opaque pointer `LLVMType` by using the `llvm_type` function:

```c
// test.c
void f(int* x) {}
```
```
// test.saw
let f_spec = do {
    val <- llvm_fresh_expanded_val (llvm_type "ptr");
    llvm_execute_func [val];
};

m <- llvm_load_module "test.bc";
ov <- llvm_verify m "f" [] false f_spec z3;
```
```
$ ~/Software/clang+llvm-16.0.0/bin/clang -g -c -emit-llvm test.c
$ ./bin/saw test.saw



[12:02:45.670] Loading file "/home/ryanglscott/Documents/Hacking/Haskell/saw-script/test.saw"
[12:02:45.673] Stack trace:
"llvm_verify" (/home/ryanglscott/Documents/Hacking/Haskell/saw-script/test.saw:8:7-8:18)
llvm_fresh_expanded_var: PtrOpaque not supported
```

I'll update this comment and file an issue about this.

Reply:
See #1879.